### PR TITLE
Send Weekly Summary CSV to Slack

### PR DESCRIPTION
**Pull request by sbowenwilliams**
image

Current columns: url, title, creator, merged, merged_by, closed_at, labels, review_comments
We'll be able to ascertain certain information, like if the emergency system was used, from the labels, but I can add an explicit boolean in the csv if that's preferable.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -2443,6 +2443,11 @@
         }
       }
     },
+    "export-to-csv": {
```
Comment (ebenoist) [inline package-lock.json:2446]: Are we using a npm or yarn in here?

File: __tests__/summary_report.test.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    let detailedPullRequest;
+    detailedPullRequest = {
+      html_url: "www.url.com",
+      title: "hi",
+      user: {login: "name"},
+      merged: 'a date',
+      merged_by: {login: "name"},
+      labels: ['label one', 'label two']
+    }
+    mocked(getClosedInPastWeek).mockResolvedValue([pullRequest]);
```
Comment (ebenoist) [inline __tests__/summary_report.test.ts:26]: Would these test be more valuable if we mocked at the API layer, or is unit the way to go here?

File: __tests__/summary_report.test.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+describe('::summaryReport', () => {
+  it('sends a csv to slack', async () => {
+    let pullRequest;
+    pullRequest = {
+      number: 1,
+    };
+
+    let detailedPullRequest;
+    detailedPullRequest = {
+      html_url: "www.url.com",
```
Comment (ebenoist) [inline __tests__/summary_report.test.ts:19]: Does this lint? Strings are inconsistently quotes (nbd -- just surprised it passes)

File: package-lock.json (hunk 1 above)
Comment (sbowenwilliams) [reply package-lock.json:2446]: npm I believe?

File: src/summary_report.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    quoteStrings: '"',
+    decimalSeparator: '.',
+    showLabels: true,
+    useTextFile: false,
+    useBom: true,
+    useKeysAsHeaders: true,
+  }
+
+  const csvExporter = new ExportToCsv(options);
+  const pullRequests = await getClosedInPastWeek();
```
Comment (ebenoist) [inline src/summary_report.ts:17]: Do we need to to catch in the unlikely case that we have an API issue?

File: src/github.ts [hunk 5]
```diff
@@ -45,6 +45,31 @@ export async function getMergedEmergencyPRsMissingReview() {
   return data.items.filter(i => i.pull_request);
 }
 
+export async function getClosedInPastWeek() {
+  let date = new Date()
+  date.setDate(date.getDate() - 7)
+  //let closedDate = `${date.getFullYear()}-${date.getMonth()}-${date.getDate()}`
```
Comment (ebenoist) [inline src/github.ts:51]: `lint`

File: package-lock.json (hunk 1 above)
Comment (ebenoist) [reply package-lock.json:2446]: Thats surprising. I think it started with yarn, but npm snuck in. Let's prefer yarn for now as we use that in all of our other js repos. Just `rm` it.

Comment (sbowenwilliams) [reply package-lock.json:2446]: rm the `package-lock` file?

File: src/summary_report.ts (hunk 4 above)
Comment (sbowenwilliams) [reply src/summary_report.ts:17]: The entire `run` method is wrapped in a catch, but I would like to report it (probably to slack) if this job in particular fails.

File: src/summary_report.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
+import * as core from '@actions/core';
+import { getClosedInPastWeek, getDetailedPR } from './github';
+import { ExportToCsv  } from 'export-to-csv';
+import { postMessage } from './slack';
+
+export async function summaryReport() {
+  const options = {
```
Comment (ebenoist) [inline src/summary_report.ts:7]: Since this is constant, you can hoist it outside of the function body

File: src/github.ts [hunk 7]
```diff
@@ -45,6 +45,30 @@ export async function getMergedEmergencyPRsMissingReview() {
   return data.items.filter(i => i.pull_request);
 }
 
+export async function getClosedInPastWeek() {
+  let date = new Date()
```
Comment (ebenoist) [inline src/github.ts:49]: This should be `const` unless you're reassigning it?

File: src/github.ts [hunk 8]
```diff
@@ -45,6 +45,30 @@ export async function getMergedEmergencyPRsMissingReview() {
...
+  date.setDate(date.getDate() - 7)
+  let closedDate = date.toISOString().substr(0, 10)
+  const { data } = await client.search.issuesAndPullRequests({
```
Comment (ebenoist) [inline src/github.ts:52]: 👍

**Comment by ebenoist** [comment]
Looking good! Thank you! Can you squash these WIP commits down and then we can get this in tomorrow?

**Comment by sbowenwilliams** [comment]
Of course!

**Pull request merged**